sdk/search/azure-search-documents/azure/search/documents/_internal/_search_index_document_batching_client.py

```diff
@@ -0,0 +1,280 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from typing import cast, List, TYPE_CHECKING
+
+import threading
+
+from azure.core.tracing.decorator import distributed_trace
+from azure.core.exceptions import HttpResponseError
+from .._api_versions import validate_api_version
+from ._utils import is_retryable_status_code
+from ._generated import SearchIndexClient
+from ._generated_serviceclient import SearchServiceClient
+from ._generated.models import IndexBatch, IndexingResult
+from ._search_documents_error import RequestEntityTooLargeError
+from ._index_documents_batch import IndexDocumentsBatch
+from .._headers_mixin import HeadersMixin
+from .._version import SDK_MONIKER
+
+if TYPE_CHECKING:
+    # pylint:disable=unused-import,ungrouped-imports
+    from typing import Any, Union
+    from azure.core.credentials import AzureKeyCredential
+
+class SearchIndexDocumentBatchingClient(HeadersMixin):
+    """A client to do index document batching.
+
+    :param endpoint: The URL endpoint of an Azure search service
+    :type endpoint: str
+    :param index_name: The name of the index to connect to
+    :type index_name: str
+    :param credential: A credential to authorize search client requests
+    :type credential: ~azure.core.credentials.AzureKeyCredential
+    :keyword int window: how many seconds if there is no changes that triggers auto flush.
+        if window is less or equal than 0, it will disable auto flush
+    :keyword int batch_size: batch size. It only takes affect when auto_flush is on
+    :keyword persistence: persistence hook. If it is set, the batch client will dump actions queue when it changes
```

Comment:
Can wwe see the type? I don't get what this is :(

Reply:
@xiangyan99 still would like some clarity on the type here

Reply:
Indeed - does a user pass this in? Would be good to see a sample or snippet on how this is used.